gitignore_parser.py

```diff
@@ -125,9 +126,14 @@ def __repr__(self):
     def match(self, abs_path: Union[str, Path]):
         matched = False
         if self.base_path:
-            rel_path = str(_normalize_path(abs_path).relative_to(self.base_path))
+            rel_path = _normalize_path(abs_path).relative_to(self.base_path).as_posix()
         else:
-            rel_path = str(_normalize_path(abs_path))
+            rel_path = _normalize_path(abs_path).as_posix()
+        # Path() strips the trailing following symbols on windows, so we need to
+        # preserve it: ' ', '.'
```

Comment:
How about we just use `PurePosixPath` instead of `Path` and special-casing Windows?

Reply:
I believe I could make this work, but it would continue to require special handling per OS due to calling `.relative_to`. 

Since Windows considers uppercase and lowercase letters the same, on a windows device `C:/home/michael` and `c:/home/michael` are the same path. However, by using `PurePosixPath`, the comparison in `relative_to` uses Posix logic, which is that uppercase and lowercase letters are different. Due to this, `.relative_to` will not find relative matches that it should on windows.

This can be seen on all test cases when running on Windows. You end up getting the error:
`ValueError: 'c:/home/michael/o.py' is not in the subpath of 'C:/home/michael' OR one path is relative and the other is absolute.`

Another issue is that `PurePosixPath` does not like `\\`. It treats it as a normal path character (not a parts separator) and thus the path parts are not actually split up as expected.

I have not done a full test, as I don't wan't to invest more time until I know this is the route you want to go, but I believe a possible solution would be to lowercase the entire path inside `_normalize_path` if we are on a windows system, it **may** get us there, but I don't garuntee it.

Something like:

```python
def _normalize_path(path: Union[str, Path]) -> PurePosixPath
    path = abspath(path).replace('\\', '/')
    if sys.platform.startswith('win'):
        path = path.lower()
    return PurePosixPath(path)
```